### ISSUE TITLE
Multipage scanning with Lexmark network MFPs?

### ISSUE THREAD

**Issue by tiramiseb** [jflesch/paperwork#534]
My new printer/scanner (Lexmark CX510de) has a strange way to manage scanning sources:

* if there is a document in the feeder, it scans the document in the feeder
* if there is no document in the feeder, it scans the document on the flatbed

... and options for scan source are "Single page" and "Multiple pages":
* with "Single page", the scan job ends automatically after scanning available documents (scanning all pages in the feeder)
* with "Multiple pages", either it does the same (with the feeder) either (where there is no document in the feeder) it scans from the flatbed and allow me to trigger a new scan or to end the job

In both cases, Paperwork only reads the first page.

(that is a really strange behaviour, but I have no choice, that's how its driver works...)

**Comment by tiramiseb** [comment]
The described behaviour is when using the scanner over the network.

When plugging the printer on the USB port, it seems to act as it should:

Sources are flatbed or feeder, the "scan" button scans only one page and the "scan from feeder" ("numériser depuis le bac") option works...

**Comment by jflesch** [comment]
Hm, let's do it the same way than for #533 . Can you post the output of the diagnostic window when connected in Ethernet please ?

**Comment by tiramiseb** [comment]
It is already connected both using USB and ethernet, both diagnostics are already included in #533...

**Comment by jflesch** [comment]
Ah my bad, I missed it.

Anyway, it is what I'm talking about in #536. Names are not standardized, and we may have really crappy list of sources.

**Comment by tiramiseb** [comment]
On this scanner, it's not even a naming problem, it's a name-to-feature assignation problem.

They use the "source" option to define the machine behaviour regarding multi-page scanning. the "multiple pages" option allow putting page after page in the flatbed scanner without moving from behind the printer.

It may be useful for big offices (this printer is made for such offices, where multiple people may want to use the hardware at the same time), allowing one person to scan multiple pages without moving back to his/her desk between two pages...

**Comment by tiramiseb** [comment]
As far as I understand this printer driver behaviour, a Sane-based software may not be able to scan multiple pages at once when the printer is networked.
The behaviour with skanlite is no different than with paperwork.

The only situation where this behaviour works is when using the software supplied by Lexmark.

**Comment by jflesch** [comment]
So .. the multi-scan dialog should work with the source "Single Page" and use the flatbed ?

**Comment by tiramiseb** [comment]
> So .. the multi-scan dialog should work with the source "Single Page" and use the flatbed ?

Nope.

I'll answer to your question in #536 here because it is related to this issue.

Plugged via USB:
* "Flatbed" + button "scan" = 1 page scanned from the flatbed
* "Flatbed" + multi-scan dialog = many pages scanned from the feeder
* "ADF" + button "scan" = 1 page scanned from the feeder
* "ADF" + multi-scan dialog = many pages scanned from the feeder

Used over the network:
* "Single Page" + button "scan" + there are pages in the feeder = all pages in the feeder are scanned and transferred to the driver, but only the first page is transmitted to paperwork
* "Single Page" + button "scan" + there is nothing in the feeder = 1 page scanned from the flatbed
"Single Page" + multi-scan dialog = nope because there is no feeder detected, the option is greyed out
"Multiple Pages" + button "scan" + there are pages in the feeder = all pages in the feeder are scanned and transferred to the driver, then the printer waits for me to put other documents (and then transfer them to the driver too) or to tell that there is no other document to scan, then only the first page is transmitted to paperwork
"Multiple Pages" + button "scan" + there is nothing in the feeder = 1 page scanned from the flatbed, then the printer waits for me to put other documents (and then transfer them to the driver too) or to tell that there is no other document to scan, then only the first page is transmitted to paperwork

**Comment by tiramiseb** [comment]
In conclusion, in my understanding, there is nothing you can do for this MFP when it is used over the network, because of how its driver works.

**Comment by jflesch** [comment]
>"Single Page" + multi-scan dialog = nope because there is no feeder detected, the option is greyed out
(...)
> there is nothing you can do for this MFP

Exception maybe ungraying the option when there is a source "Multiple pages" :)

**Comment by jflesch** [comment]
So to sum up:

* I must ungray the option for multiple scans.
* When "single page" is selected in the settings + when using the multi-scan dialog, I'm still wondering if I should switch automagically to "multiple pages". I guess not, since it seems this scanner is made to scan many pages at once using the Flatbed.

**Comment by tiramiseb** [comment]
I don't think ungraying the multiple scan option with regular flatbed scanners would make sense. It may be very confusing.

I also don't think you should modify paperwork's behaviour to make it work with a setup which is obviously made for big offices, where there probable are no paperwork users.

**Comment by jflesch** [comment]
You misunderstood me : Currently, the option is ungrayed if the scanner has a source called "ADF" or "Automatic Document Feeder" or "Feeder". I just need to add "Multiple pages" to the list.

Regarding the use of the multi-scan dialog + "Single page", I guess this is a rare corner case and actually nobody will really use it, so I can just discard it for now, and switch to "Multiple pages" automatically.

**Comment by tiramiseb** [comment]
> I just need to add "Multiple pages" to the list.

Nope, there is absolutely no relation.

"Multiple pages" only means « the printer must ask me to add pages after the ones that have been scanned », but a scan job with the "Multiple pages" source results in one-page scans in Sane.

Once again, this weird Lexmark driver does not transmit more than the first page to sane.

**Comment by jflesch** [comment]
Hmmm, weird ; are you sure ? Because with the button "scan", Paperwork requests only one page, so this is actually normal.

**Comment by tiramiseb** [comment]
full procedure with the "Single page" source:
1/ click on "scan" in ANY software (tested with paperwork and skanlite)
2/ the driver window pops up, telling the user to go to the printer, put their document in the scanner and trigger the job (called "profile")
3/ go in front of the printer
4/ put 3 sheets in the feeder tray
5/ trigger the profile
6/ the feeder takes the 3 sheets one after another
7/ the printer screen goes back to the home screen
8/ in paperwork or scanlite, only the first page is displayed

full procedure with the "Multiple pages" source:
1/ click on "scan" in ANY software (tested with paperwork and skanlite)
2/ the driver window pops up, telling the user to go to the printer, put their document in the scanner and trigger the job (called "profile")
3/ go in front of the printer
4/ put 3 sheets in the feeder tray
5/ trigger the profile
6/ the feeder takes the 3 sheets one after another
7/ the printer screen tells either to put new documents in the scanner or to finish the profile
8/ put 3 more sheets
9/ tell the printer to continue scanning
10/ the feeder takes the 3 sheets one after another
11/ the printer screen tells either to put new documents in the scanner or to finish the profile
12/ tell the printer to end the profile
13/ the printer screen goes back to the home screen
14/ in paperwork or scanlite, only the first page is displayed

**Comment by jflesch** [comment]
> the driver window pops up

There is a X11 window integrated to the Sane driver ... how .. ugly.

**Comment by tiramiseb** [comment]
with the "multiple pages scan" option in paperwork, paperwork doesn't request one job with multiple pages to the scanner: it triggers a new scan job for every scanned page... so the "multiple pages" lexmark option is really not related.

**Comment by tiramiseb** [comment]
> > the driver window pops up
> There is a X11 window integrated to the Sane driver ... how .. ugly.

Without a window, how would you:
* tell the user (s)he can go to the printer room and trigger the scan job ?
* allow the user to cancel the job remotely ?


(by the way, it is not only a X11 window, it's java!)

**Comment by jflesch** [comment]
> tell the user (s)he can go to the printer room and trigger the scan job ?

The user has to go to the scanner anyway, and there is a screen on the scanner, no?

> allow the user to cancel the job remotely ?

There is a function "sane_cancel()" (Paperwork doesn't expose it yet)

Anyway, based on what you're telling, I won't change anything.
However, if you want to try the mutli-scan dialog with "Multiple pages" (just to be safe ;), you can:
* change the source in the settings
* close Paperwork
* edit ~/.config/paperwork.conf and set has_feeder to True
* restart Paperwork

**Comment by tiramiseb** [comment]
> The user has to go to the scanner anyway, and there is a screen on the scanner, no?

So you mean the paperwork/skanlite window would sit there, not displaying anything, letting the user guess what he must do on the scanner ?

Oh, by the way, they planned this feature for offices where multiple people would need scanning at the same time: there may be multiple jobs (profiles) waiting on the MFP's scanner, and the java window tells the user which one is his. See screenshot below:

![lexmark_network_scan_window](https://cloud.githubusercontent.com/assets/1292007/20604049/bacfa328-b264-11e6-9c47-9069b8e4f99d.png)

> There is a function "sane_cancel()"

The need is not to cancel the Sane job. The driver needs to tell the printer itself to delete the profile from its internal queue.

----

This complicated procedure (great for big offices but annoying for me, I have the printer next to me in my personal office) is the main reason why I will not use the scanner over the network :)

**Comment by jflesch** [comment]
Ok, I see your point.

> The need is not to cancel the Sane job. The driver needs to tell the printer itself to delete the profile from its internal queue.

In this case, that's the same thing. Canceling the Sane job *is* removing it from the printer's queue.

**Comment by jflesch** [comment]
I'm wondering if it would make sense to display first something like "Waiting for scanner" in Paperwork while waiting for the first bytes to arrive from the scanner.
Because currently, I guess having the red dot moving when no scan is being done must be kind of ugly .

**Comment by tiramiseb** [comment]
> Canceling the Sane job is removing it from the printer's queue. (or it should)

Nope. It is removing it from the sane's queue. In most cases, a scanner does not have a queue : sane requests a scan, the scanner does its job.

From the software point of view, this driver simulates a scanner with a very long reaction time...

**Comment by tiramiseb** [comment]
> I'm wondering if it would make sense to display first something like "Waiting for scanner" in Paperwork while waiting for the first bytes to arrive from the scanner.

That never bothered me...

**Comment by tiramiseb** [comment]
I think this issue can be closed as wontfix...

**Comment by jflesch** [comment]
Since you seem sure no change is actually required,  I guess so.
(eh, less work for me ;)

**Comment by tiramiseb** [comment]
I'm sure no useful change is possible...
